webapp/components/survey/NodeDefDetails/ButtonBar.js

```diff
@@ -31,18 +33,18 @@ const ButtonBar = (props) => {
             ? dispatch(
                 DialogConfirmActions.showDialogConfirm({
                   key: 'common.cancelConfirm',
-                  onOk: NodeDefsActions.cancelNodeDefEdits(history),
+                  onOk: Actions.cancelNodeDefEdits(history),
                 })
               )
-            : dispatch(NodeDefsActions.cancelNodeDefEdits(history))
+            : dispatch(Actions.cancelNodeDefEdits(history))
         }
       >
         {i18n.t(dirty ? 'common.cancel' : 'common.back')}
       </button>
       <button
         type="button"
         className="btn btn-primary"
-        onClick={() => dispatch(NodeDefsActions.saveNodeDefEdits())}
+        onClick={() => dispatch(Actions.saveNodeDefEdits())}
```

Comment:
Actions is a local store object. dispatch is the redux dispatch. 
how come these two are connected?

Reply:
yes, but saveNodeDefEdits needs to dispatch other actions too, how can it do it otherwise?